dlt/common/json/__init__.py

```diff
@@ -116,19 +140,24 @@ def custom_pua_encode(obj: Any) -> str:
     elif isinstance(obj, bytes):
         return _B64BYTES + base64.b64encode(obj).decode("ascii")
     elif hasattr(obj, "asdict"):
-        return obj.asdict()  # type: ignore
+        return obj.asdict()  # type: ignore[no-any-return]
```

Comment:
these ignores can be more narrow now, as we allow a broader return value (dicts)